### PR TITLE
Ts/4079 task optimize cleanup 3

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description

This deals with correctly updating state
This also converts a lot of string into enums in prep for pyciemss-service plumbing PR

**Done here:**
-Cannot optimize on an intervention that have multiple interventions grouped inside.
-do not provide things that we are already providing to these group forms.
-correct emits to properly update state.
-do not allow for `param_value_and_start_time` it is not ready yet on pyciemss-service and wont be for awhile.
-Rename from: `startTimeOption and newValueOption` to: `objectiveFunctionOption` 
-`objectiveFunctionOption` now has an `enum` not type `string`.
-Remove not required defaults from static component

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue [hunk 1]
```diff
@@ -591,7 +587,7 @@ const runOptimize = async () => {
 			numSamples: knobs.value.numSamples,
 			maxiter: knobs.value.maxiter,
 			maxfeval: knobs.value.maxfeval,
-			alpha: (100 - props.node.state.constraintGroups[0].riskTolerance) / 100,
+			alpha: props.node.state.constraintGroups[0].riskTolerance / 100,
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue:590]: Could you add a small comment explaining why we need to divde the `riskTolerance` by 100 for `alpha`?

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue [hunk 2]
```diff
@@ -51,27 +69,58 @@
...
-								<tera-input type="number" v-model="knobs[objective]" />
-							</div>
+						<div class="label-and-input">
+							<label>Start time</label>
+							<tera-input
+								type="number"
+								v-model="knobs.startTime"
+								@update:model-value="emit('update-self', knobs)"
+							/>
+						</div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue:108]: @blanchco added a `label` props to `tera-input`.
Also do we need those `<div class="lable-and-input">`?

```suggestion
						<div class="label-and-input">
							<tera-input
								type="number"
								:label="Start time"
								v-model="knobs.startTime"
								@update:model-value="emit('update-self', knobs)"
							/>
						</div>
```

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue [hunk 3]
```diff
@@ -97,51 +146,48 @@ import { StaticIntervention } from '@/types/Types';
...
 
 const emit = defineEmits(['update-self']);
 
-const knobs = ref({
-	isActive: props.config.isActive ?? false,
+const staticInterventions = ref<StaticIntervention[]>(
+	props.config.intervention.staticInterventions
+);
+
+const knobs = ref<InterventionPolicyGroupForm>({
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue:163]:
```suggestion
const knobs = ref<InterventionPolicyGroupForm>({ ...props.config});
```

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue [hunk 4]
```diff
@@ -97,51 +146,48 @@ import { StaticIntervention } from '@/types/Types';
...
 
-const NEW_VALUE_OPTIONS = ['lower bound', 'upper bound', 'initial guess'];
-const START_TIME_OPTIONS = ['earliest', 'latest', 'inital guess'];
-
-const newValueInputs = ['lowerBound', 'upperBound', 'initialGuess'];
-const newStartTimeInputs = ['startTime', 'endTime', 'startTimeGuess'];
+const isNotEditable = computed(() => {
+	if (staticInterventions.value.length === 1) return false;
+	return true;
+});
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue:179]:
```suggestion
const isNotEditable = computed(() => staticInterventions.value.length !== 1);
```

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue [hunk 5]
```diff
@@ -51,27 +69,58 @@
...
-								<label :for="objective">{{ NEW_VALUE_OPTIONS[index] }}</label>
-								<tera-input type="number" v-model="knobs[objective]" />
-							</div>
+						<div class="label-and-input">
+							<label>Lower bound</label>
+							<tera-input
+								type="number"
+								v-model="knobs.lowerBoundValue"
+								@update:model-value="$emit('update-self', knobs)"
+							/>
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue:78]: - `tera-input` has a label parameter, you may want to check if it is more suitable here.
- Also you may want to use type=nist

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue (hunk 1 above)
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue:590]: Just checking the removal of `(100 - blah)` is intentional?

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue (hunk 3 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue:163]: Thanks!

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue (hunk 1 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss.vue]: correct! Nelson has rewritten the sentence structure for this change

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue (hunk 5 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-static-intervention-policy-group.vue]: Good catch

**Review by YohannParis — APPROVED** [review]

**Review by asylves1 — APPROVED** [review]

**Pull request merged**